rules/gcp_audit_rules/gcp_dns_zone_modified_or_deleted.yml

```diff
@@ -0,0 +1,249 @@
+AnalysisType: rule
+Description: Detection for GCP DNS zones that are deleted, patched, or updated.
+DisplayName: "GCP DNS Zone Modified or Deleted"
+Enabled: true
+Filename: gcp_dns_zone_modified_or_deleted.py
+Runbook: Verify that this modification or deletion was expected. These operations are high-impact events and can result in downtimes or total outages.
+Severity: Low
+Tests:
+    - ExpectedResult: true
+      Log:
+        insertid: -nkgd1se1zsiw
+        logName: projects/detection-testing-380018/logs/cloudaudit.googleapis.com%2Factivity
+        protoPayload:
+            at_sign_type: type.googleapis.com/google.cloud.audit.AuditLog
+            authenticationInfo:
+                principalEmail: staging@pantherstaging.io
+            authorizationInfo:
+                - granted: true
+                  permission: dns.managedZones.delete
+                  resourceAttributes: {}
+            methodName: dns.managedZones.delete
+            request:
+                '@type': type.googleapis.com/cloud.dns.api.ManagedZonesDeleteRequest
+                managedZone: test-zone
+                project: detection-testing-380018
+            requestMetadata:
+                callerIP: 98.156.138.170
```

Comment:
```suggestion
                callerIP: 12.12.12.12
```

Reply:
Good call!

Reply:
Cleaned up some of the logs in [`2efec4e` (#779)](https://github.com/panther-labs/panther-analysis/pull/779/commits/2efec4ebc7bfb23813b6f025ff68da42d0cce0da).